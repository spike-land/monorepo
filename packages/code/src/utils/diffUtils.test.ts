import { describe, expect, it } from "vitest";
import { extractDiffContent, isDiffContent } from "./diffUtils";

describe("diffUtils", () => {
  describe("extractDiffContent", () => {
    it("should extract original and modified content correctly", () => {
      const diffContent = `
<<<<<<< SEARCH
Original content
=======
Modified content
>>>>>>> REPLACE
      `;

      const result = extractDiffContent(diffContent);

      expect(result).toEqual({
        original: "Original content",
        modified: "Modified content",
      });
    });

    it("should handle empty original content", () => {
      const diffContent = `
<<<<<<< SEARCH
=======
Modified content
>>>>>>> REPLACE
      `;

      const result = extractDiffContent(diffContent);

      expect(result).toEqual({
        original: "",
        modified: "Modified content",
      });
    });

    it("should handle empty modified content", () => {
      const diffContent = `
<<<<<<< SEARCH
Original content
=======
>>>>>>> REPLACE
      `;

      const result = extractDiffContent(diffContent);

      expect(result).toEqual({
        original: "Original content",
        modified: "",
      });
    });

    it("should handle multi-line content", () => {
      const diffContent = `
<<<<<<< SEARCH
Original
content
with
multiple
lines
=======
Modified
content
with
multiple
lines
>>>>>>> REPLACE
      `;

      const result = extractDiffContent(diffContent);

      expect(result).toEqual({
        original: "Original\ncontent\nwith\nmultiple\nlines",
        modified: "Modified\ncontent\nwith\nmultiple\nlines",
      });
    });

    it("should return empty strings for invalid diff content", () => {
      const invalidContent = "This is not a valid diff content";

      const result = extractDiffContent(invalidContent);

      expect(result).toEqual({
        original: "",
        modified: "",
      });
    });
  });

  describe("isDiffContent", () => {
    it("should return true for valid diff content", () => {
      const diffContent = "<<<<<<< SEARCH\nOriginal\n=======\nModified\n>>>>>>> REPLACE";
      expect(isDiffContent(diffContent)).toBe(true);
    });

    it("should return false for invalid diff content", () => {
      const invalidContent = "This is not a valid diff content";
      expect(isDiffContent(invalidContent)).toBe(false);
    });

    it("should do partial", () => {
      const diffContent = `
      <<<<<<< SEARCH
      Original conte`;

      const result = isDiffContent(diffContent);
      expect(result).toBe(true);
      expect(extractDiffContent(diffContent).original).toBe("Original conte");
      expect(extractDiffContent(diffContent).modified).toBe("");
    });
  });

  it("should do partial2", () => {
    const diffContent = `
    <<<<<<< SEARCH
    Original conte
    =======
    z
    `;

    const result = isDiffContent(diffContent);
    expect(result).toBe(true);
    expect(extractDiffContent(diffContent).original).toBe("Original conte");
    expect(extractDiffContent(diffContent).modified).toBe("z");
  });
});
