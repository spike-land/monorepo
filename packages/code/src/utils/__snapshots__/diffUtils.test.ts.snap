// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`diffUtils > should do partial5 1`] = `
"<<<<<<< SEARCH
    <div className='w-full font-mono text-sm rounded-lg overflow-hidden shadow-lg bg-white'>
      <div className='bg-gray-200 px-4 py-2 flex items-center space-x-2'>
        <div className='w-3 h-3 rounded-full bg-red-500'></div>
        <div className='w-3 h-3 rounded-full bg-yellow-500'></div>
        <div className='w-3 h-3 rounded-full bg-green-500'></div>
      </div>
      <div className='bg-gray-100 text-gray-800 py-2 px-4 text-sm border-b border-gray-300 flex justify-between items-center'>
=======
    <div className='w-full font-mono text-sm rounded-lg overflow-hidden shadow-lg bg-gray-900'>
      <div className='bg-gray-800 text-gray-200 py-2 px-4 text-sm border-b border-gray-700 flex justify-between items-center'>
>>>>>>> REPLACE

<<<<<<< SEARCH
        style={{
          ...tomorrow,
          'pre[class*="language-"]': {
            ...tomorrow['pre[class*="language-"]'],
            background: "#f8f8f8",
            padding: "1rem",
          },
          'code[class*="language-"]': {
            ...tomorrow['code[class*="language-"]'],
            color: "#333",
          },
        }}
=======
        style={{
          ...tomorrow,
          'pre[class*="language-"]': {
            ...tomorrow['pre[class*="language-"]'],
            background: "#1a202c",
            padding: "1rem",
          },
          'code[class*="language-"]': {
            ...tomorrow['code[class*="language-"]'],
            color: "#e2e8f0",
          },
        }}
>>>>>>> REPLACE

<<<<<<< SEARCH
import { ClipboardIcon } from "@heroicons/react/24/outline";
=======
import { ClipboardIcon, DocumentDuplicateIcon, ClipboardDocumentIcon } from "@heroicons/react/24/outline";
>>>>>>> REPLACE

<<<<<<< SEARCH
  const [copied, setCopied] = useState(false);
=======
  const [copied, setCopied] = useState(false);
  const [iconIndex, setIconIndex] = useState(0);
  const icons = [ClipboardIcon, DocumentDuplicateIcon, ClipboardDocumentIcon];
  const Icon = icons[iconIndex];
>>>>>>> REPLACE

<<<<<<< SEARCH
        <div className='relative'>
          <button
            className='text-gray-600 hover:text-gray-800 ml-2 transition-colors duration-200'
            onClick={() => {
              navigator.clipboard.writeText(value);
              setCopied(true);
              setTimeout(() => setCopied(false), 2000);
            }}>
            <ClipboardIcon className='w-5 h-5' />
          </button>
          {copied && (
            <div className='absolute right-0 mt-2 py-1 px-2 bg-gray-800 text-white text-xs rounded shadow-lg'>
              Copied!
            </div>
          )}
        </div>
=======
        <div className='relative'>
          <button
            className='text-gray-300 hover:text-white ml-2 transition-colors duration-200'
            onClick={() => {
              navigator.clipboard.writeText(value);
              setCopied(true);
              setTimeout(() => setCopied(false), 2000);
              setIconIndex((iconIndex + 1) % icons.length);
            }}>
            <Icon className='w-5 h-5' />
          </button>
          {copied && (
            <div className='absolute right-0 mt-2 py-1 px-2 bg-gray-700 text-white text-xs rounded shadow-lg'>
              Copied!
            </div>
          )}
        </div>
>>>>>>> REPLACE

<<<<<<< SEARCH

<div className='w-full font-mono text-sm rounded-lg overflow-hidden shadow-lg bg-white'>
<div className='bg-gray-200 px-4 py-2 flex items-center space-x-2'>
<div className='w-3 h-3 rounded-full bg-red-500'></div>
<div className='w-3 h-3 rounded-full bg-yellow-500'></div>
<div className='w-3 h-3 rounded-full bg-green-500'></div>
</div>
<div className='bg-gray-100 text-gray-800 py-2 px-4 text-sm border-b border-gray-300 flex justify-between items-center'>

======= 

<div className='w-full font-mono text-sm rounded-lg overflow-hidden shadow-lg bg-gray-900'>
<div className='bg-gray-800 text-gray-200 py-2 px-4 text-sm border-b border-gray-700 flex justify-between items-center'>

>>>>>>> REPLACE

<<<<<<< SEARCH

style={{
...tomorrow,
'pre[class*="language-"]': {
...tomorrow['pre[class*="language-"]'],
background: "#f8f8f8",
padding: "1rem",
},
'code[class*="language-"]': {
...tomorrow['code[class*="language-"]'],
color: "#333",
},
}}

======= 

style={{
...tomorrow,
'pre[class*="language-"]': {
...tomorrow['pre[class*="language-"]'],
background: "#1a202c",
padding: "1rem",
},
'code[class*="language-"]': {
...tomorrow['code[class*="language-"]'],
color: "#e2e8f0",
},
}}

>>>>>>> REPLACE

<<<<<<< SEARCH

import { ClipboardIcon } from "@heroicons/react/24/outline";

======= 

import { ClipboardIcon, DocumentDuplicateIcon, ClipboardDocumentIcon } from "@heroicons/react/24/outline";

>>>>>>> REPLACE

<<<<<<< SEARCH

const [copied, setCopied] = useState(false);

======= 

const [copied, setCopied] = useState(false);
const [iconIndex, setIconIndex] = useState(0);
const icons = [ClipboardIcon, DocumentDuplicateIcon, ClipboardDocumentIcon];
const Icon = icons[iconIndex];

>>>>>>> REPLACE

<<<<<<< SEARCH

<div className='relative'>
<button
className='text-gray-600 hover:text-gray-800 ml-2 transition-colors duration-200'
onClick={() => {
navigator.clipboard.writeText(value);
setCopied(true);
setTimeout(() => setCopied(false), 2000);
}}>
<ClipboardIcon className='w-5 h-5' />
</button>
{copied && (
<div className='absolute right-0 mt-2 py-1 px-2 bg-gray-800 text-white text-xs rounded shadow-lg'>
Copied!
</div>
)}
</div>

======= 

<div className='relative'>
<button
className='text-gray-300 hover:text-white ml-2 transition-colors duration-200'
onClick={() => {
navigator.clipboard.writeText(value);
setCopied(true);
setTimeout(() => setCopied(false), 2000);
setIconIndex((iconIndex + 1) % icons.length);
}}>
<Icon className='w-5 h-5' />
</button>
{copied && (
<div className='absolute right-0 mt-2 py-1 px-2 bg-gray-700 text-white text-xs rounded shadow-lg'>
Copied!
</div>
)}
</div>

>>>>>>> REPLACE"
`;
